### PR TITLE
update: connection approval declined response message

### PR DESCRIPTION
**Pull request by NoelStephensUnity**
**Select the type of change:**

- [ ] Small Changes - Typos, formatting, slight revisions
- [X] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site and Docusaurus

**Purpose of the Pull Request:**
Adding documentation for the new `NetworkManager.ConnectionApprovalResponse.Reason` property along with an example of how it is used.

### CODE REVIEW

File: docs/basics/connection-approval.md
Comment (jeffreyrainy) [inline docs/basics/connection-approval.md:68]: I'm not sure about the default value. Is this a case where we get support tickets "Strange reason pops up even though the server did not provide it" ?

File: docs/basics/connection-approval.md
Comment (armstrongl) [inline docs/basics/connection-approval.md:29]: - **PlayerPrefabHash**: The type of player prefab to use for the authorized player. It uses the default `NetworkManager` defined player prefab if it's null.
- **Position** and **Rotation**: The position and rotation of the player when spawned.
- **Pending**: Provides the ability to mark the approval as "pending" to delay the authorization until other user-specific code finishes the approval process.
- **Reason**: If `Approved` is `false`, you can populate the `Reason` with a `string`-based message (or JSON) explaining why the client wasn't approved.

File: docs/basics/connection-approval.md
Comment (armstrongl) [inline docs/basics/connection-approval.md:31]: Unlike previous versions of Netcode for GameObjects, where users had to invoke a callback within the connection approval handler method, users now only need to set the appropriate properties of the `NetworkManager.ConnectionApprovalResponse` class. Part of this update allows users to set their `ConnectionApprovalResponse` to `Pending`, which provides users additional time to process any other tasks involved with the player approval process.

File: docs/basics/connection-approval.md
Comment (armstrongl) [inline docs/basics/connection-approval.md:79]: If you need to deny a player from connecting for any reason (such as reaching the maximum number of connections or an invalid authorization), you can use the optional `NetworkManager.ConnectionApprovalResponse.Reason` property on the `NetworkManager.ConnectionApprovalResponse` structure. When `NetworkManager.ConnectionApprovalResponse.Approved` is false and `NetworkManager.ConnectionApprovalResponse.Reason` has been populated with the reason for denying the player's request to connect, the server sends the client a `DisconnectReasonMessage`. When the client receives the `DisconnectReasonMessage`, the `NetworkManager.DisconnectReason` property populates with the `NetworkManager.ConnectionApprovalResponse.Reason` message. The example below shows how this works:

File: docs/basics/connection-approval.md
Comment (armstrongl) [inline docs/basics/connection-approval.md:125]: The `ConnectionApprovalRequest.Payload` parameter takes any custom data the client should send to the server. The data is usually a matchmaking ticket, room password, or something similar that determines if a connection should be approved or not. The `connectionData` is specified on the client-side in the `NetworkingConfig` supplied when connecting.

File: docs/basics/connection-approval.md
Comment (armstrongl) [inline docs/basics/connection-approval.md:211]: To populate the `AlternatePlayerPrefabs` list:
1. Open the scene containing the in-scene placed `NetworkObject` to which the `ConnectionApprovalCallback` is attached.
2. Find each alternate player prefab you wish to add to the list, select the prefab (but don't open it), and copy the `GlobalObjectIdHash` value by right-clicking and selecting **Copy**.

**Review by armstrongl — APPROVED** [review]
I know many of my suggestions are for content that wasn't added in this PR, so don't worry about them too much. I can always open a PR later to make style guide corrections. It LGTM 😀

**Pull request merged**
